### PR TITLE
Feature/control outdated packages

### PR DESCRIPTION
**Pull request by lasote**
Full implementation of #572

### CODE REVIEW

File: conans/model/info.py [hunk 1]
```diff
@@ -193,6 +194,7 @@ def loads(text):
         result.full_options = OptionsValues.loads(parser.full_options)
         result.full_requires = RequirementsList.loads(parser.full_requires)
         result.requires = RequirementsInfo(result.full_requires, None)
+        result.recipe_hash = parser.recipe_hash
```
Comment (memsharded) [inline conans/model/info.py]: ConfigParser returns empty string "" for non existing fields, maybe None is preferred here to indicate absence

File: conans/client/recipe_hash.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    for root, _, files in os.walk(folder):
+        for name in files:
+            if not name.endswith(".pyc"):
+                paths.append(os.path.join(root, name))
+    paths = sorted(paths)
+    digest_lines = "\n".join([_get_file_digest(path) for path in paths])
+    return sha1(digest_lines.encode("utf-8"))
+
+
+def hashes_match(recipe_hash, package_recipe_hash):
```
Comment (memsharded) [inline conans/client/recipe_hash.py]: boilerplate?

File: conans/client/recipe_hash.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+def _get_file_digest(path):
+    return sha1(_normalize_crlfs(path))
+
+
+def get_normalized_hash(folder):
+    paths = []
+    for root, _, files in os.walk(folder):
+        for name in files:
+            if not name.endswith(".pyc"):
```
Comment (memsharded) [inline conans/client/recipe_hash.py]: From Manifest:
```python
file_dict.pop(".DS_Store", None)  # Exclude tmp in mac
```
seems that such file have to be ignored from the hash. In fact the full implementation is:

```python
from conans.paths import CONAN_MANIFEST, CONANFILE
        file_dict.pop(PACKAGE_TGZ_NAME, None)  # Exclude the PACKAGE_TGZ_NAME
        file_dict.pop(EXPORT_TGZ_NAME, None)  # Exclude the EXPORT_TGZ_NAME
        file_dict.pop(CONAN_MANIFEST, None)  # Exclude the MANIFEST itself
        file_dict.pop(CONANFILE + "c", None)  # Exclude the CONANFILE.pyc
        file_dict.pop(".DS_Store", None)  # Exclude tmp in mac

        file_dict = {key: value for key, value in file_dict.items()
                     if not key.startswith("__pycache__")}
```

File: conans/test/command/search_test.py [hunk 4]
```diff
@@ -193,6 +195,8 @@ def package_search_properties_filter_test(self):
             arch: x86
             compiler: gcc
             compiler.version: 4.3
+        [recipe hash]
```
Comment (memsharded) [inline conans/test/command/search_test.py]: I would say that this is a bit verbose and uninformative. What can I do with such hash?. What does it means?. I'd say that a single line something like: ``outdated from recipe: True/False`` could be better.

File: conans/client/recipe_hash.py (hunk 3 above)
Comment (memsharded) [inline conans/client/recipe_hash.py]: Conversely, mabye the manifest should exclude the recipe hash too?

File: conans/test/command/search_test.py (hunk 4 above)
Comment (lasote) [inline conans/test/command/search_test.py]: I thought that it could be useful to compare between generated packages. But it's not very important.

**Pull request merged**